Sources/WebSocketKit/WebSocket.swift

```diff
@@ -256,6 +310,37 @@ public final class WebSocket {
         }
     }
 
+    private func proceedEventData(_ socket: WebSocket, _ text: String) -> Bool {
+        guard !events.isEmpty, let data = text.data(using: .utf8) else { return false }
+        do {
+            let prototype = try jsonDecoder.decode(WebSocketEventPrototype.self, from: data)
+            if let bind = events.first(where: { $0.0 == prototype.event }) {
+                bind.value(socket, data)
+                return true
+            }
+        } catch {
+            logger.trace("Unable to decode incoming event cause it doesn't conform to `WebSocketEventPrototype` model: \(error)")
+        }
+        return false
+    }
+
+    private func proceedEventData(_ socket: WebSocket, _ byteBuffer: ByteBuffer) -> Bool {
+        guard !events.isEmpty, byteBuffer.readableBytes > 0 else { return false }
+        do {
+            var bytes: [UInt8] = byteBuffer.getBytes(at: byteBuffer.readerIndex, length: byteBuffer.readableBytes) ?? []
+            let data = Data(bytes: &bytes, count: byteBuffer.readableBytes)
+
+            let prototype = try jsonDecoder.decode(WebSocketEventPrototype.self, from: data)
+            if let bind = events.first(where: { $0.0 == prototype.event }) {
+                bind.value(socket, data)
+                return true
+            }
+        } catch {
+            logger.trace("Unable to decode incoming event cause it doesn't conform to `WebSocketEventPrototype` model: \(error)")
```

Comment:
Nit spelling
```suggestion
            logger.trace("Unable to decode incoming event because it doesn't conform to `WebSocketEventPrototype` model: \(error)")
```